liblwgeom/lwgeom.c

```diff
@@ -465,6 +465,76 @@ lwgeom_release(LWGEOM *lwgeom)
 
 }
 
+/* free lwgeom and all underlying objects except serialized_pointlist */
```

Comment:
It looks like it's not just the serialized_pointlist but possibly more things ?
For once I see the array of pointers to underlying geometries in the COLLECTION case, and then I wonder about the BBOX cache, sometimes it points to underlying buffer from serialized version and sometimes it's created anew, I think such difference is encoded in flags somehow.

Reply:
lwgeom_release deep leaves serialized_pointlist not freed, everything else including array of pointers is freed.
bbox cache would have to be handled correctly by lwgeom_free, right? I don't see anything specific about bbox in lwgeom_free so I made release_deep to just free memory if pointer is not null.

the same purpose for which I introduced lwgeom_release_deep can be achived by lwgeom_free but then readonly flag should be set to 1 before a call so serialized_pointlist won't be freed.
https://github.com/postgis/postgis/blob/177de9ca1c1f4fa6cc57271c5810bcc26ebade84/liblwgeom/ptarray.c#L323
not sure if it is a better way to go.